reactive/jenkins.py

```diff
@@ -230,3 +248,34 @@ def update_nrpe_config(nagios):
 def stop():
     service_stop("jenkins")
     status_set("maintenance", "Jenkins stopped")
+
+
+@when("reset.jenkins.storage")
+def reset_jenkins_storage():
+    status_set("maintenance", "Configuring Jenkins Storage")
+
+    jenkins_installed = get_state("apt.installed.jenkins")
+    if jenkins_installed:
+        service_stop('jenkins')
+
+    storage_dir = unitdata.kv().get("storage.dir")
+    if storage_dir is None:
+        log("Setting Jenkins to use local storage")
+        paths.Storage().unlink_home()
+    else:
+        log("Setting Jenkins to use storage at {}".format(storage_dir))
+        paths.Storage().link_home(storage_dir)
+
+    if jenkins_installed:
+        status_set("maintenance", "Restarting Jenkins")
+        service_start('jenkins')
+        Service().check_ready()
+
+    remove_state("reset.jenkins.storage")
+
+    if get_state('jenkins.bootstrapped'):
+        # JENKINS_HOME just changed trigger bootstrap again
+        remove_state("jenkins.bootstrapped")
+        bootstrap_jenkins()
+    else:
+        status_set('active', 'Ready')
```

Comment:
Hm, I now realize the issue a bit better. Essentially from the user perspective the Jenkins deployment would be reset from scratch in the case of "if get_state('jenkins.bootstrapped')", as we discussed on IRC, if I recall correctly.

I'm starting to think that if you don't have a concrete use case for attaching storage *after* initial deployment, I'd prefer that we ignore the attach/detach storage events completely for now, and perhaps set the status to some error code (e.g. "Attaching storage after deployment not supported").

Alternatively, I'd tend to change my mind and say that we should copy existing data to the new directory (and do that in a separate PR). Otherwise, the behavior is a bit surprising: as I user I prefer to be told that some action is required rather than having my Jenkins apparently reset (I'd have to know that the data has been saved in .bak).

Sorry for the back and forth, and feel free to ping me again on IRC.